packages/nns/src/token.ts

```diff
@@ -0,0 +1,100 @@
+import { ICPTs } from "../proto/ledger_pb";
+import { E8S_PER_TOKEN, TOKEN_DECIMAL_ACCURACY } from "./constants/constants";
+import { FromStringToTokenError } from "./enums/token.enums";
+
+export const convertStringToE8s = (
+  amount: string
+): bigint | FromStringToTokenError => {
+  // Remove all instances of "," and "'".
+  amount = amount.trim().replace(/[,']/g, "");
+
+  // Verify that the string is of the format 1234.5678
+  const regexMatch = amount.match(/\d*(\.\d*)?/);
+  if (!regexMatch || regexMatch[0] !== amount) {
+    return FromStringToTokenError.InvalidFormat;
+  }
+
+  const [integral, fractional] = amount.split(".");
+
+  let e8s = BigInt(0);
+
+  if (integral) {
+    try {
+      e8s += BigInt(integral) * E8S_PER_TOKEN;
+    } catch {
+      return FromStringToTokenError.InvalidFormat;
+    }
+  }
+
+  if (fractional) {
+    if (fractional.length > 8) {
+      return FromStringToTokenError.FractionalMoreThan8Decimals;
+    }
+    try {
+      e8s += BigInt(fractional.padEnd(8, "0"));
+    } catch {
+      return FromStringToTokenError.InvalidFormat;
+    }
+  }
+
+  return e8s;
+};
+
+export type Token = {
```

Comment:
for consistency reason maybe an `interface` (looks like so far we mostly have `interfarce`)?

Reply:
This is a lint rule, shall we also add it?

Not now obviously, hehe.

Reply:
Sure if possible later on why not!

Don't misunderstand me, no strong opinion on type or interface. Just noticed we have many interface notably in the converters so thought about the consistency

Reply:
I am not aware or any consistency here nor in nns-dapp hehe. I don't remember on agreeing on one or the other.

But I am a fan of consistency, so maybe we want to add something.

Reply:
done